Sources/ArcGISToolkit/Components/Offline/OfflineMapAreasView.swift

```diff
@@ -28,10 +28,19 @@ public struct OfflineMapAreasView: View {
     /// A Boolean value indicating whether the preplanned map areas are being reloaded.
     @State private var isReloadingPreplannedMapAreas = false
     
+    /// The currently selected map.
+    private var selectedMap: Binding<Map?>
+    
     /// Creates an `OfflineMapAreasView` with a given web map.
-    /// - Parameter map: The web map.
-    public init(map: Map) {
-        _mapViewModel = StateObject(wrappedValue: MapViewModel(map: map))
+    /// - Parameters:
+    ///   - onlineMap: The web map.
+    ///   - selectedMap: A binding to the currently selected map.
+    public init(
+        onlineMap: Map,
+        selectedMap: Binding<Map?>
+    ) {
+        _mapViewModel = StateObject(wrappedValue: MapViewModel(map: onlineMap))
```

Comment:
From the [Initialize state objects using external data](https://developer.apple.com/documentation/swiftui/stateobject#Initialize-state-objects-using-external-data) section of the doc:

> Use caution when doing this. SwiftUI only initializes a state object the first time you call its initializer in a given view. This ensures that the object provides stable storage even as the view’s inputs change. However, it might result in unexpected behavior or unwanted side effects if you explicitly initialize the state object.

If you haven't already or you haven't done so recently, please read through that section of the doc and consider whether any other changes need to be made.

Reply:
I'm not sure the `online` map is expected to change after the `OfflineMapAreasView` is declared. The view model uses the map to access it's portal item so that changes on arcgis online will show in the view. Should changing the `online` map parameter be supported?

Reply:
Since this view is presented modally, most likely the online map won't change while it is presented.